lua/ulx/modules/sh/fun.lua

```diff
@@ -994,14 +994,19 @@ end
 hook.Add( "PlayerDisconnected", "ULXMaulDisconnectedCheck", maulDisconnectedCheck, HOOK_MONITOR_HIGH )
 
 ------------------------------ Strip ------------------------------
-function ulx.stripweapons( calling_ply, target_plys )
+function ulx.stripweapons( calling_ply, target_plys, wep )
 	for i=1, #target_plys do
-		target_plys[ i ]:StripWeapons()
+		if wep and target_plys[ i ]:HasWeapon(wep) then
```

Comment:
If they don't have the weapon you're asking to be stripped of, it'll remove all your weapons. I doubt you need to make sure they have the weapon, though it probably doesn't hurt to test